# Trabajo Final: Análisis de Misiones Espaciales (1957-2022)
Curso: Herramientas básicas para el Análisis de Datos  
Comisión: 999201624  
Profesores: Virginia Marich, Santiago Marich  
Alumno: Tobias Martinez  


## 1. Introducción y Justificación del Dataset
El presente proyecto analiza misiones espaciales para estudiar costos, resultados y evolución del sector.

Fue desarrollado en el marco de un curso de análisis de datos, incorporando exploración y conclusiones propias.

La industria aeroespacial representa uno de los mayores referentes en materia de eficiencia y gestión del riesgo, es por ello, que he seleccionado un dataset que documenta las misiones espaciales desde el inicio de la carrera espacial hasta la actualidad.

## 2. Pregunta Inicial de Investigación
El objetivo es analizar la evolución del sector espacial a lo largo del tiempo, enfocandose en tres pilares clave:

1.  ¿Se ha vuelto el acceso al espacio más económico con el desarrollo de nuevas tecnologías?
2.  ¿Existe una relacion entre el presupuesto invertido y el resultado de la misión?
3. ¿Existe un cambio en la estructura del sector espacial en términos de participación pública y privada?

## 3. Datos acerca del dataset
*   Link: https://mavenanalytics.io/data-playground/space-missions
### Glosario
* **Company**: Compañía responsable
* **Location**: Ubicación del despegue
* **Date**: Fecha del despegue
* **Time**: Hora del despegue (UTC)
* **Rocket**: Nombre del cohete utilizado
* **Mission**: Nombre de la misión.
* **RocketStatus**: Estado del cohete a Agosto de 2022 (Activo/Inactivo)
* **Price**: Costo del cohete en millones de dolares
* **MissionStatus**: Estado de la mision (Éxito, Fracaso, Fracaso parcial, Fracaso en el pre-lanzamiento)

## 4. Preparación del Entorno y Carga de Datos
En esta sección, importamos las librerías que vamos a utilizar y conectamos Google Drive para acceder al dataset.

In [ ]:
import pandas as pd
import numpy as np
import plotly.express as px
from plotly.subplots import make_subplots
import plotly.graph_objects as go
#Aclaración: Estuve probando los gráficos con sns y ninguno me gustó más que usar los interactivos de Plotly.

#Conectamos la ruta del dataset
df = pd.read_csv('/content/drive/MyDrive/Curso/TrabajoFinal/MisionesEspaciales/space_missions.csv', sep=',', encoding='latin1', on_bad_lines='skip')

#Hacemos una prueba rapida que nos muestre de que la conexion es correcta.
df.head()

,Company,Location,Date,Time,Rocket,Mission,RocketStatus,Price,MissionStatus
0,RVSN USSR,"Site 1/5, Baikonur Cosmodrome, Kazakhstan",1957-10-04,19:28:00,Sputnik 8K71PS,Sputnik-1,Retired,NaN,Success
1,RVSN USSR,"Site 1/5, Baikonur Cosmodrome, Kazakhstan",1957-11-03,02:30:00,Sputnik 8K71PS,Sputnik-2,Retired,NaN,Success
2,US Navy,"LC-18A, Cape Canaveral AFS, Florida, USA",1957-12-06,16:44:00,Vanguard,Vanguard TV3,Retired,NaN,Failure
3,AMBA,"LC-26A, Cape Canaveral AFS, Florida, USA",1958-02-01,03:48:00,Juno I,Explorer 1,Retired,NaN,Success
4,US Navy,"LC-18A, Cape Canaveral AFS, Florida, USA",1958-02-05,07:33:00,Vanguard,Vanguard TV3BU,Retired,NaN,Failure


## 5. Exploración inicial


### Visualizar primeras 7 filas

In [ ]:
df.head(7)

,Company,Location,Date,Time,Rocket,Mission,RocketStatus,Price,MissionStatus
0,RVSN USSR,"Site 1/5, Baikonur Cosmodrome, Kazakhstan",1957-10-04,19:28:00,Sputnik 8K71PS,Sputnik-1,Retired,NaN,Success
1,RVSN USSR,"Site 1/5, Baikonur Cosmodrome, Kazakhstan",1957-11-03,02:30:00,Sputnik 8K71PS,Sputnik-2,Retired,NaN,Success
2,US Navy,"LC-18A, Cape Canaveral AFS, Florida, USA",1957-12-06,16:44:00,Vanguard,Vanguard TV3,Retired,NaN,Failure
3,AMBA,"LC-26A, Cape Canaveral AFS, Florida, USA",1958-02-01,03:48:00,Juno I,Explorer 1,Retired,NaN,Success
4,US Navy,"LC-18A, Cape Canaveral AFS, Florida, USA",1958-02-05,07:33:00,Vanguard,Vanguard TV3BU,Retired,NaN,Failure
5,AMBA,"LC-26A, Cape Canaveral AFS, Florida, USA",1958-03-05,18:27:00,Juno I,Explorer 2,Retired,NaN,Failure
6,US Navy,"LC-18A, Cape Canaveral AFS, Florida, USA",1958-03-17,12:15:00,Vanguard,Vanguard 1,Retired,NaN,Success


###Estructura del dataset

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4630 entries, 0 to 4629
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company        4630 non-null   object
 1   Location       4630 non-null   object
 2   Date           4630 non-null   object
 3   Time           4503 non-null   object
 4   Rocket         4630 non-null   object
 5   Mission        4630 non-null   object
 6   RocketStatus   4630 non-null   object
 7   Price          1265 non-null   object
 8   MissionStatus  4630 non-null   object
dtypes: object(9)
memory usage: 325.7+ KB


Vemos que tenemos tipos de datos que no poseen el formato que esperamos, como la columna *Price*, la cual está en tipo **cualitativo** y necesitamos que su dato refleje uno de tipo **cuantitativo**.

### Cantidad de filas y columnas

In [ ]:
df.shape #Utilizamos esta función para determinar la cantidad de filas/registros (4630) Y columnas (9)

(4630, 9)

### Registros nulos

In [ ]:
# Verificamos cuántos valores nulos hay por columna
print(df.isnull().sum())

Company             0
Location            0
Date                0
Time              127
Rocket              0
Mission             0
RocketStatus        0
Price            3365
MissionStatus       0
dtype: int64


Vemos que tenemos 127 registros vacíos en la columna de *Time* y 3365 registros en la columna de *Price*. Ahora bien, para el caso práctico de este análisis, creo pertinente eliminar los registros vacíos que contenga la columna *Price*, esto debido a que lo que se está evaluando es un factor económico del dataset e imputar este dato sería inferir en exceso en el resultado. Con respecto a la columna *Time*, *Rocket*, *Mission* y *RocketStatus* no son relevantes para dar respuesta a las preguntas planteadas, por lo que las eliminaré.

### Registros duplicados

In [ ]:
df.duplicated().sum()

np.int64(1)

### Numeros con comas y puntos

In [ ]:
df['Price'].unique()
#Esta función nos permite visualizar rapidamente que algunos registros cuentan con comas como separador y otros con puntos.
#Deberemos unificar el criterio para poder trabajarlos correctamente.

array([nan, '63.23', '59', '1,160.00', '450', '30.8', '5,000.00', '35',
       '136.6', '40', '29.75', '25', '69.7', '45', '20', '29', '64.68',
       '41.8', '47', '90', '109', '164', '200', '133', '135', '190',
       '17.42', '350', '130', '153', '7', '115', '21', '29.15', '123',
       '20.14', '31', '112.5', '120', '59.5', '140', '46', '30', '145',
       '80', '37', '65', '55', '56.5', '62', '100', '19', '5.3', '15',
       '85', '5.8', '7.5', '48.5', '39', '23', '67', '12', '28.3', '2.5',
       '4.9', '27'], dtype=object)

Finalizado este bloque de exploración inicial podemos empezar a limpiar y estructurar la información en base a nuestras necesidades.

## 6. Limpieza y estructuración

### Duplicados

In [ ]:
df.drop_duplicates(inplace=True) #Eliminamos los registros duplicados. Con el parámetro inplace lo hacemos de manera permanente.

### Estandarización de fechas, categorias y unidades.

In [ ]:
df['Price'] = df['Price'].astype(str).str.replace(',', '') #Quitamos las comas
df['Price'] = pd.to_numeric(df['Price'], errors='coerce') #Aplicamos formato númerico
df.tail(8) #Vista rápida para ver como va quedando

,Company,Location,Date,Time,Rocket,Mission,RocketStatus,Price,MissionStatus
4622,SpaceX,"LC-39A, Kennedy Space Center, Florida, USA",2022-07-15,00:44:00,Falcon 9 Block 5,CRS SpX-25,Active,67.00,Success
4623,CASC,"LC-9, Taiyuan Satellite Launch Center, China",2022-07-15,22:57:00,Long March 2C,SuperView Neo 2-01 & 02,Active,30.80,Success
4624,SpaceX,"SLC-40, Cape Canaveral SFS, Florida, USA",2022-07-17,14:20:00,Falcon 9 Block 5,Starlink Group 4-22,Active,67.00,Success
4625,SpaceX,"SLC-4E, Vandenberg SFB, California, USA",2022-07-22,17:39:00,Falcon 9 Block 5,Starlink Group 3-2,Active,67.00,Success
4626,CASC,"LC-101, Wenchang Satellite Launch Center, China",2022-07-24,06:22:00,Long March 5B,Wentian,Active,NaN,Success
4627,SpaceX,"LC-39A, Kennedy Space Center, Florida, USA",2022-07-24,13:38:00,Falcon 9 Block 5,Starlink Group 4-25,Active,67.00,Success
4628,CAS Space,"Jiuquan Satellite Launch Center, China",2022-07-27,04:12:00,Zhongke-1A,Demo Flight,Active,NaN,Success
4629,CASC,"LC-3, Xichang Satellite Launch Center, China",2022-07-29,13:28:00,Long March 2D,Yaogan 35 Group 03,Active,29.75,Success


A su vez, solo utilizaremos el año de la columna *Date*. Para ello, primero debemos darle formato de fecha y luego crear una columna llamada *Year*, donde se albergará unicamente el año de la misión.

In [ ]:
# Convertimos el texto a formato fecha
df['Date'] = pd.to_datetime(df['Date'], errors='coerce')

# Creamos la columna año
df['Year'] = df['Date'].dt.year
df.tail(8)

,Company,Location,Date,Time,Rocket,Mission,RocketStatus,Price,MissionStatus,Year
4622,SpaceX,"LC-39A, Kennedy Space Center, Florida, USA",2022-07-15,00:44:00,Falcon 9 Block 5,CRS SpX-25,Active,67.00,Success,2022
4623,CASC,"LC-9, Taiyuan Satellite Launch Center, China",2022-07-15,22:57:00,Long March 2C,SuperView Neo 2-01 & 02,Active,30.80,Success,2022
4624,SpaceX,"SLC-40, Cape Canaveral SFS, Florida, USA",2022-07-17,14:20:00,Falcon 9 Block 5,Starlink Group 4-22,Active,67.00,Success,2022
4625,SpaceX,"SLC-4E, Vandenberg SFB, California, USA",2022-07-22,17:39:00,Falcon 9 Block 5,Starlink Group 3-2,Active,67.00,Success,2022
4626,CASC,"LC-101, Wenchang Satellite Launch Center, China",2022-07-24,06:22:00,Long March 5B,Wentian,Active,NaN,Success,2022
4627,SpaceX,"LC-39A, Kennedy Space Center, Florida, USA",2022-07-24,13:38:00,Falcon 9 Block 5,Starlink Group 4-25,Active,67.00,Success,2022
4628,CAS Space,"Jiuquan Satellite Launch Center, China",2022-07-27,04:12:00,Zhongke-1A,Demo Flight,Active,NaN,Success,2022
4629,CASC,"LC-3, Xichang Satellite Launch Center, China",2022-07-29,13:28:00,Long March 2D,Yaogan 35 Group 03,Active,29.75,Success,2022


### Registros nulos

In [ ]:
#Eliminamos los registros vacios de la columna Price. Con el parámetro inplace lo hacemos de manera permanente.
df.dropna(subset=['Price'], inplace=True)
print(df.isnull().sum())

Company          0
Location         0
Date             0
Time             6
Rocket           0
Mission          0
RocketStatus     0
Price            0
MissionStatus    0
Year             0
dtype: int64


### Eliminación de columnas

In [ ]:
# Eliminamos la columnas que no son necesarias.
df.drop(columns=['Time', 'Rocket', 'Mission', 'RocketStatus'], inplace=True)# Con el parámetro inplace lo hacemos de manera permanente.

## 7. Visualización

####¿Promedio o Mediana? ¿Que conviene más?


In [ ]:
# Costos por cada resultado de misión
stats_by_status = df.groupby('MissionStatus')['Price'].describe()
print(stats_by_status)

                    count        mean         std   min   25%    50%      75%  \
MissionStatus                                                                   
Failure              48.0   55.532083   90.695538   2.5   7.5  37.00   49.375   
Partial Failure      19.0  131.168421  260.971680  25.0  40.0  47.00   67.350   
Prelaunch Failure     1.0   62.000000         NaN  62.0  62.0  62.00   62.000   
Success            1196.0  131.316856  261.063992   2.5  30.8  64.68  123.000   

                      max  
MissionStatus              
Failure             450.0  
Partial Failure    1160.0  
Prelaunch Failure    62.0  
Success            5000.0  


El dataset analizado presenta una estructura de costos muy dispar, caracterizada por un pequeño volumen de misiones de altísimo presupuesto frente a una gran mayoría de lanzamientos de menor escala. En este contexto, el promedio se ve fuertemente sesgado por esos valores atípicos (outliers), lo que distorsiona la visión de lo que cuesta un lanzamiento típico. Es por ello que la mediana resulta una medida mucho más fiel para entender la realidad del sector, ya que refleja el costo de la mayor parte de las operaciones, compuestas por satélites y cohetes más pequeños. Es por ello que, a lo largo de esta visualización, trabajaremos con la mediana para capturar correctamente donde la eficiencia y la frecuencia han desplazado a los modelos de gasto extraordinario del siglo pasado.

###1) ¿Se ha vuelto el acceso al espacio más económico con el desarrollo de nuevas tecnologías?

In [ ]:
# 1. Preparar datos con MEDIANA en lugar de PROMEDIO
missions_trend = df.groupby('Year').size().reset_index(name='Misiones')
# Utilizamos la función .median() para capturar la mediana
median_price_year = df.groupby('Year')['Price'].median().reset_index(name='Precio_Mediano')
combined_df = pd.merge(missions_trend, median_price_year, on='Year', how='left')
# 2. Variable numérica para la tendencia
x = np.arange(len(combined_df))
# Tendencia de las misiones
coef_misiones = np.polyfit(x, combined_df['Misiones'], 1)
trend_misiones = np.poly1d(coef_misiones)(x)
# Tendencia del precio
coef_precio = np.polyfit(x, combined_df['Precio_Mediano'], 1)
trend_precio = np.poly1d(coef_precio)(x)
# 3. Gráfico
fig = make_subplots(specs=[[{"secondary_y": True}]])
# Precio
fig.add_trace(
    go.Scatter(
        x=combined_df['Year'],
        y=combined_df['Precio_Mediano'],
        name="Mediana de Precio (MDD)",
        line=dict(color='rgba(100, 149, 237, 1)')
    ),
    secondary_y=True,
)
# Tendencia precio
fig.add_trace(
    go.Scatter(
        x=combined_df['Year'],
        y=trend_precio,
        name="Tendencia Precio (Mediana)",
        line=dict(color='rgba(100, 149, 237, 1)', dash='dash')
    ),
    secondary_y=True,
)
# Misiones reales
fig.add_trace(
    go.Scatter(
        x=combined_df['Year'],
        y=combined_df['Misiones'],
        name="Cantidad de Misiones",
        mode='lines+markers',
        line=dict(color='white')
    ),
    secondary_y=False,
)
# Tendencia misiones
fig.add_trace(
    go.Scatter(
        x=combined_df['Year'],
        y=trend_misiones,
        name="Tendencia Misiones",
        line=dict(color='white', dash='dash')
    ),
    secondary_y=False,
)
# Layout ajustado
fig.update_layout(
    title_text="Evolución y Tendencias: Volumen vs. Mediana de Costos",
    template='plotly_dark',
    legend=dict(orientation="h", yanchor="bottom", y=1.02, xanchor="right", x=1)
)
fig.update_xaxes(title_text="Año")
fig.update_yaxes(title_text="Cantidad de Misiones", secondary_y=False)
fig.update_yaxes(title_text="Mediana de Precio (Millones USD)", secondary_y=True)
fig.show()

####Interpretación breve de Evolución de Misiones y su Costo Promedio (MDD)
Basado en el análisis, podemos distinguir una tendencia alcista en el volumen de misiones y una marcada reducción del costo en los últimos años, lo que refleja no solo un abaratamiento real gracias a la reutilización de cohetes, otras tecnologías y el ingreso del sector privado.

###2) ¿Existe una relacion entre el presupuesto invertido y el resultado de la misión?

In [ ]:
# Creamos el gráfico de dispersión de los costos por resultado de misiones.
fig = px.scatter(
    df,
    x='MissionStatus',
    y='Price',
    color='MissionStatus',
    title='Relación entre Presupuesto (MDD) y Resultado de la Misión',
    labels={'MissionStatus': 'Estado de la misión', 'Price': 'Precio (Millones USD)'},
    hover_data=['Company', 'Year'],
    template='plotly_dark'
)

# Ajustamos la opacidad y el tamaño
fig.update_traces(marker=dict(size=20, opacity=0.6))

fig.show()

####Interpretación breve de la relación entre presupuestos y resultados
A partir de la mediana podemos entender que las misiones exitosas tienden a tener un mayor presupuesto que las fallidas.

* Exitosas: 64.68 MDD
* Fallas parciales: 47 MDD
* Fallidas: 37 MDD
* Fallidas antes del despegue: 62 MDD (Solo 1 caso, baja representatividad)

A mayor inversión, mayor nivel de pruebas, validaciones y redundancias. Esto reduce el riesgo de fallo, sin embargo, **no lo garantiza**.

###3) ¿Existe un cambio en la estructura del sector espacial en términos de participación pública y privada?

In [ ]:
# Nueva columna llamada Decade agrupando los años por década
df['Decade'] = (df['Year'] // 10) * 10

# Agrupar por década y empresa, contando la cantidad de misiones
top_companies_by_decade = df.groupby(['Decade', 'Company']) \
                            .size() \
                            .reset_index(name='Misiones')
# Ordenar por década y cantidad de misiones (de mayor a menor)
# Luego tomar las 5 empresas con más misiones en cada década
top_5_by_decade = top_companies_by_decade \
                    .sort_values(['Decade', 'Misiones'], ascending=[True, False]) \
                    .groupby('Decade') \
                    .head(5)
# Gráfico de barras
fig = px.bar(
    top_5_by_decade,
    x='Decade',
    y='Misiones',
    color='Company',
    title='Top 5 Empresas con más Misiones por Década',
    barmode='group',  # Barras agrupadas por década
    template='plotly_dark'
)
fig.show()

####Interpretación breve de la reestructuracion del sector espacial
El gráfico muestra una tendencia al cambio progresivo en la composición del sector. En las primeras décadas dominaban actores gubernamentales como la NASA, RVSN USSR/ROSCOSMOS. A partir de 1990 aparecen mas competidores y desde los años 2000 crece con fuerza el sector privado, con empresas como SpaceX o Arianespace. Sin embargo, las agencias estatales no pierden peso, como lo evidencia el CASC, donde aumenta exponencialmente su participación.

## 8. Conclusiónes y cierre

### Conclusión del proceso de análisis
Durante el desarrollo del trabajo se avanzó desde una exploración inicial del dataset hacia un análisis más estructurado, combinando limpieza y formateo de datos combinando visualizaciones para responder las preguntas planteadas.

Algunos de los desafíos que se presentaron:

* Valores faltantes en la columna Price
* Formatos inconsistentes en datos numéricos
* Existencia de outliers que distorsionaban métricas

Como se abordaron estos desafíos:
*   Limpieza y filtrado de datos
*   Uso de métricas más adecuadas como la mediana
* Análisis segmentado por características para explotar mejor la información.



### Conclusión general del tema abordado:
El análisis permitió confirmar que el acceso al espacio se ha vuelto más accesible en términos económicos y más abierto a la participacion del sector, generando un entorno más competitivo frente a las grandes agencias gubernamentales. Sin embargo, el éxito sigue siendo un desafío que no depende únicamente del nivel de inversión.